File: packages/dart/lib/Seguranca/Ataque.dart
import 'package:criptografia/Seguranca/AlvoCriptografico.dart';
import 'package:criptografia/Seguranca/ResultadoAtaque.dart';

/// Contrato de um ataque da suíte.
abstract class Ataque {
  String nome();

  /// Roda o ataque contra o alvo e retorna o resultado.
  /// Deve lançar [SkipAtaqueException] se o alvo não suportar os recursos
  /// necessários (ex: não expõe gerarKeystreamBruto).
  ResultadoAtaque executar(AlvoCriptografico alvo);
}

File: packages/julia/src/Seguranca/Ataque.jl
"""
Tipo abstrato de um ataque.

Cada ataque concreto é um `struct ... <: Ataque` que define `nome(a)` e
`executar(a, alvo)`. `executar` deve lançar `SkipAtaqueException` se o alvo
não suportar os recursos necessários.
"""
abstract type Ataque end

function nome end
function executar end
